presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -296,6 +314,14 @@ protected Type visitCurrentTime(CurrentTime node, StackableAstVisitorContext<Con
         @Override
         protected Type visitSymbolReference(SymbolReference node, StackableAstVisitorContext<Context> context)
         {
+            if (context.getContext().isInLambda()) {
```

Comment:
`StackableAstVisitorContext` contains information (path) about AST nodes from the current node up to the root. I think you could leverege that instead of writing and your own context.

I could image that as:
```
if (context.isIn(LambdaExpression.class)) {
````